demo/DemoFull.tsx

```diff
@@ -80,23 +81,29 @@ function Demo() {
                                 } else {
                                     setEditorValid('valid');
                                 }
-                                setEditorText(JSON.parse(event.target.value))
+                                setEditorText(JSON.parse(event.target.value));
                             }}
-                        /> 
+                        />
                     </Grid>
-                : null}
+                    : null}
                 <Grid item aria-label='altgosling component' xs={12}>
-                    { selectedExample === 'editor' ? 
-                        (editorValid === 'invalid' ? 
-                            <Typography variant='body1' color='error'>No Gosling or AltGosling components could be loaded.</Typography>
-                        : <AltGoslingComponent spec={editorText} download name={selectedExample} />
-                        )
-                    : <AltGoslingComponent spec={examples[selectedExample]} download name={selectedExample} />
-                    }
+                    {(() => {
+                        let goslingSpec: string;
+                        if (selectedExample === 'editor' && editorValid === 'invalid') {
+                            return <Typography variant='body1' color='error'>No Gosling or AltGosling components could be loaded.</Typography>;
+                        } else if (selectedExample === 'editor' && editorValid !== 'invalid') {
+                            goslingSpec = editorText;
+                        } else {
+                            goslingSpec = examples[selectedExample];
+                        }
+                        return <AltGoslingComponent spec={goslingSpec} download name={selectedExample} onAltGoslingSpecUpdate={spec => {
+                            console.log('AltGosling Spec Updated', spec);
+                        }} />;
+                    })()}
```

Comment:
This is the only part I edited to include `onAltGoslingSpecUpdate`.

Reply:
Changed the structure a bit to make it more readable and avoid duplicated <AltGoslingComponent/> use